sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -40,56 +40,93 @@ public typealias CommunicationSignalingErrorHandler = (CommunicationSignalingErr
 /// TrouterTokenRefreshHandler for fetching tokens.
 internal typealias TrouterTokenRefreshHandler = (_ stopSignalingClient: Bool, Error?) -> Void
 
-class CommunicationSignalingClient {
-    private var selfHostedTrouterClient: SelfHostedTrouterClient
+public class CommunicationSignalingClient {
+    private var selfHostedTrouterClient: SelfHostedTrouterClient?
     private var communicationSkypeTokenProvider: CommunicationSkypeTokenProvider
-    private var trouterUrlRegistrar: TrouterUrlRegistrar
+    private var trouterUrlRegistrar: TrouterUrlRegistrar?
     private var logger: ClientLogger
     private var communicationHandlers: [ChatEventId: CommunicationHandler] = [:]
+    private var configuration: RealTimeNotificationConfiguration?
 
+    // Step 1: Initialize with basic properties
     init(
         communicationSkypeTokenProvider: CommunicationSkypeTokenProvider,
         logger: ClientLogger = ClientLoggers.default(tag: "AzureCommunicationSignalingClient")
     ) throws {
         self.logger = logger
         self.communicationSkypeTokenProvider = communicationSkypeTokenProvider
+    }
 
-        let trouterSkypeTokenHeaderProvider = TrouterSkypetokenAuthHeaderProvider(
-            skypetokenProvider: communicationSkypeTokenProvider
-        )
-
-        let trouterSettings = try getTrouterSettings(token: communicationSkypeTokenProvider.token)
+    // Step 2: Configure with fetched TrouterSettings
+    public func configure(token: String, endpoint: String, completionHandler: @escaping (Result<Void, AzureError>) -> Void) {
+        self.getTrouterSettings(token: token, endpoint: endpoint) { result in
+            switch result {
+            case .success(let configuration):
+                self.configuration = configuration
+
+                let trouterSkypeTokenHeaderProvider = TrouterSkypetokenAuthHeaderProvider(
+                    skypetokenProvider: self.communicationSkypeTokenProvider
+                )
+                
+                // Remove the "https://" prefix from the URLs
+                let trouterHostname = configuration.trouterServiceUrl.replacingOccurrences(of: "https://", with: "")
+                let registrarUrlWithoutScheme = configuration.registrarServiceUrl.replacingOccurrences(of: "https://", with: "")
+                
+                // Remove the "v3/registrations" or "V3/registrations" suffix from the regsitrar URL
+                let registrarBasePath = registrarUrlWithoutScheme
```

Comment:
Please ask Trouter team to understand more about the Trouter service URL

Reply:
Based on Jiri's suggestion, we should remove the "/v3/a" suffixes from both Trouter endpoint and Registrar endpoint on service side and let client side to append the values. 

Updated.